application/forms/EditNodeForm.php

```diff
@@ -443,4 +474,14 @@ public function isValid($data)
 
         return parent::isValid($data);
     }
+
+    public function shouldBeDeleted()
+    {
+        if ($this->deleteButtonName === null) {
+            return false;
+        }
+
+        $name = $this->deleteButtonName;
```

Comment:
This looks like misusing the property to me.
Please do the following instead:
```suggestion
$deleteButton = $this->getElement($this->deleteButtonName);

if ($deleteButton === null) {
    return false;
}

return $this->getSentValue($deleteButton->getName()) === $deleteButton->getLabel();